### PR TITLE
Migrate `extern-flag-pathless`, `silly-file-names`, `metadata-dep-info`, `cdylib-fewer-symbols` and `symbols-include-type-name` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

`cdylib-fewer-symbols` demands a Windows try-job. (Almost guaranteed to fail, but 7 years is a long time)

try-job: x86_64-gnu-distcheck
try-job: x86_64-msvc
try-job: aarch64-apple

### CODE REVIEW

**Comment by Oneirical** [comment]
Well, how cool is that? I think it's because `llvm-readobj` is more platform-agnostic than nm. Ignore statement removed.

File: tests/run-make/cdylib-fewer-symbols/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+fn main() {
+    // Compile a cdylib
+    rustc().input("foo.rs").run();
+    let out = llvm_readobj().arg("--symbols").input(dynamic_lib_name("foo")).run().stdout_utf8();
+    let out = // All hidden symbols must be removed.
+        out.lines().filter(|&line| !line.trim().contains("HIDDEN")).collect::<Vec<_>>().join("\n");
+    assert!(!&out.contains("__rdl_"));
+    assert!(!&out.contains("__rde_"));
+    assert!(!&out.contains("__rg_"));
+    assert!(!&out.contains("__ruse_"));
```
Comment (Kobzol) [inline tests/run-make/cdylib-fewer-symbols/rmake.rs:18]:
```suggestion
    assert!(!&out.contains("__rust_"));
```
Your *ruse* was almost successful :)

File: tests/run-make/extern-flag-pathless/rmake.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// It is possible, since #64882, to use the --extern flag without an explicit
+// path. In the event of two --extern flags, the explicit one with a path will take
+// priority, but otherwise, it is a more concise way of fetching specific libraries.
+// This test checks that the default priority of explicit extern flags and rlibs is
+// respected.
+// See https://github.com/rust-lang/rust/pull/64882
+
+use run_make_support::{dynamic_lib_name, fs_wrapper, run, run_fail, rust_lib_name, rustc};
```
Comment (Kobzol) [inline tests/run-make/extern-flag-pathless/rmake.rs:8]: Shouldn't this be `//@ ignore-cross-compile`? We run the compiled code.

File: tests/run-make/cdylib-fewer-symbols/rmake.rs (hunk 1 above)
Comment (Oneirical) [reply tests/run-make/cdylib-fewer-symbols/rmake.rs]: Foiled yet again! Fixed.

**Comment by Kobzol** [comment]
Thanks!

@bors r+ rollup=iffy

**Comment by ehuss** [comment]
@bors r-

bors sync

**Comment by Oneirical** [comment]
Failed due to the `ignore-cross-compile` being removed, where the `__rust` symbol appears in the output. Ignore restored and reason added. @rustbot review

**Comment by jieyouxu** [comment]
Going to run one last battery of try jobs, then r=Kobzol if try jobs come back green.

**Comment by jieyouxu** [comment]
Sorry I missed this ping somehow!
@bors r=@Kobzol

**Comment by jieyouxu** [comment]
Thanks!
@bors r=@Kobzol

**Pull request merged**